src/Apps/Settings/Routes/Purchases/Components/SettingsPurchasesRow.tsx

```diff
@@ -240,20 +298,7 @@ const SettingsPurchasesRow: FC<React.PropsWithChildren<SettingsPurchasesRowProps
               name={artwork?.partner?.name ?? ""}
             />
 
-            {order.displayState === "PAYMENT_FAILED" && (
-              <Button
-                // @ts-ignore
-                as={RouterLink}
-                to={`/orders/${order.internalID}/payment/new`}
-                variant="primaryBlack"
-                size="large"
-                width="50%"
-                minWidth={["240px", "200px"]}
-                my={[1, 0]}
-              >
-                Update Payment Method
-              </Button>
-            )}
+            {getOrderActionButton(displayState, order.internalID)}
```

Comment:
Re: long comment above, this is where a component rendering would look better than a `getSomethingMightBeNull()` function call.